### PR TITLE
Added checkstyle for "Methods starting with Capital Letters" 

### PR DESCRIPTION
**Pull request by hpandeycodeit**
Added checkstyle for "Methods starting with Capital Letters" and changed the method names violating this.

Issue #6936

### CODE REVIEW

**Comment by hpandeycodeit** [comment]
These failing asserts aren't part of my commit. Shall I fix them?

File: extensions-core/kafka-indexing-service/src/test/java/org/apache/druid/indexing/kafka/KafkaIndexTaskTest.java [hunk 1]
```diff
@@ -697,6 +697,23 @@ public void testIncrementalHandOffMaxTotalRows() throws Exception
     }
     final Map<Integer, Long> nextOffsets = ImmutableMap.copyOf(task.getRunner().getCurrentOffsets());
 
+      // Check metrics
+    Assert.assertEquals(8, task.getRunner().getRowIngestionMeters().getProcessed());
```
Comment (clintropolis) [inline extensions-core/kafka-indexing-service/src/test/java/org/apache/druid/indexing/kafka/KafkaIndexTaskTest.java]: It looks like maybe some logic got shuffled around, could this be related to failing tests?

Comment (hpandeycodeit) [reply extensions-core/kafka-indexing-service/src/test/java/org/apache/druid/indexing/kafka/KafkaIndexTaskTest.java]: yeah looks like they got moved earlier in the code. I have repositioned them and tests as passing on my local. Hopefully, this should build fine. :)

**Comment by clintropolis** [comment]
Teamcity failure looks related, but it's not really this PR's fault. I think some of the method renaming picks up as duplicate code between 2 of the Kafka tests. I think refactoring that isn't really related to this PR.

**Comment by hpandeycodeit** [comment]
@clintropolis It's pointing to lines : 889 and 2216 which I don't see in the code ? Which code it is pointing to exactly ? May be I can fix in a different PR?

**Comment by clintropolis** [comment]
>@clintropolis It's pointing to lines : 889 and 2216 which I don't see in the code ? Which code it is pointing to exactly ? May be I can fix in a different PR?

Oh yeah, I guess those numbers don't line up so much, I assumed it was complaining about [this part](https://github.com/apache/incubator-druid/pull/7118/files#diff-a4eac18bddd0caa9fbe7683631db48c6R69) which also exists in `KafkaIndexTaskTest.java`... I'll re-run teamcity and see if it happens again 🙃

**Comment by hpandeycodeit** [comment]
oh okay. looks like this time it's showing different errors.

**Comment by clintropolis** [comment]
>oh okay. looks like this time it's showing different errors.

It's probably just teamcity being flakey, it's been having some issues lately. I'll try to retrigger a few times to see if it will pass, else I'll just go ahead and merge this.

**Comment by hpandeycodeit** [comment]
@clintropolis So it passed finally :) Thanks for looking into this.

**Comment by hpandeycodeit** [comment]
Thank you @leventov @clintropolis

**Pull request merged**